### PR TITLE
first part of implementation of Users API

### PR DESCRIPTION
**Pull request by isabelsang**
- Using the Users API to log users in/out.
- New servlet checks if user is logged in/out and sends json containing login status info (whether user is logged in, user's email, URL to log in/out) back to the client 
- JS fetches from this servlet and uses the status info to 
      - hide the comment form if user is logged out and display a login hyperlink
      - display the comment form if user is logged in and display a logout hyperlink
- Still need to:
       - style login/out hyperlink
       - have the comments display the user's email or nickname

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      UserService userService = UserServiceFactory.getUserService();
+      boolean isUserLoggedIn = userService.isUserLoggedIn();
+
+      Map<String, String> loginStatus = new HashMap<String, String>();
+      loginStatus.put("isUserLoggedIn", String.valueOf(isUserLoggedIn));
+      
+      if (isUserLoggedIn) {
+        loginStatus.put("email", userService.getCurrentUser().getEmail());
+
+        String urlToRedirectToAfterUserLogsOut = "/";
```
Comment (BearBurg) [inline portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:43]: This variable can be extracted as a field.

File: portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        loginStatus.put("url", userService.createLogoutURL(urlToRedirectToAfterUserLogsOut));
+      } else {
+        loginStatus.put("email", null);
+
+        String urlToRedirectToAfterUserLogsIn = "/";
```
Comment (BearBurg) [inline portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:48]: same here.

File: portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        loginStatus.put("url", userService.createLoginURL(urlToRedirectToAfterUserLogsIn));
+    }
+
+    Gson gson = new Gson();
```
Comment (BearBurg) [inline portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:52]: We can extract it as a field. for example: `private static final Gson gson = new Gson();`

File: portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java [hunk 4]
```diff
@@ -28,6 +28,10 @@
 
 @WebServlet("/login-status")
 public class LoginStatusServlet extends HttpServlet {
+  
+  private static final Gson gson = new Gson();
+  private static final String urlToRedirectToAfterUserLogsOut = "/";
```
Comment (BearBurg) [inline portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:33]: Please follow the constant names convention. For example: `URL_TO_REDIRECT_TO_AFTER_USER_LOG_OUT`
https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names

Comment (isabelsang) [reply portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:33]: Added this change to PR #18

**Pull request merged**